src/main/kotlin/Utilities/JsonGenerator.kt

```diff
@@ -1,14 +1,20 @@
 package Utilities
 
+import com.google.gson.Gson
 import org.json.JSONObject
 
 class JsonGenerator {
     private val jsonObj = JSONObject()
+    private val gson = Gson()
 
     fun generateJsonString(mapOfKeysToValues: Map<String, Any>): String {
         for (entries in mapOfKeysToValues) {
             jsonObj.put(entries.key, entries.value)
         }
         return jsonObj.toString()
     }
+
+    fun resourceToJson(resource: Any): String {
```

Comment:
I think `Any` is often not recommended in typed languages. Is there nothing more specific that you could use instead?

Reply:
I believe I can set this function up as a generic function and define some generic constraints so multiple types are explicit, but this is something I need to do further research into.